macros/edr/tests/on_run_end/handle_tests_results.sql

```diff
@@ -335,22 +338,25 @@
         'test_params': elementary.insensitive_get_dict_value(test_node, 'test_params'),
         'severity': elementary.insensitive_get_dict_value(test_node, 'severity'),
         'status': elementary.insensitive_get_dict_value(run_result_dict, 'status'),
-        'failures': elementary.insensitive_get_dict_value(run_result_dict, 'failures')
+        'failures': elementary.insensitive_get_dict_value(run_result_dict, 'failures'),
+        'test_results_sample': elementary.render_test_sample(elementary.get_test_sample(test_node))
     }%}
     {{ return(test_result_dict) }}
 {% endmacro %}
 
-{% macro get_test_result_rows_as_dicts(flatten_test_node, sample_limit = none) %}
-    {% set test_row_dicts = [] %}
-    {% set test_compiled_code = elementary.get_compiled_code(flatten_test_node) %}
-    {% if test_compiled_code %}
-        {% if sample_limit %}
-            {% set test_compiled_code = test_compiled_code ~ ' limit ' ~ sample_limit %}
-        {% endif %}
-        {% set test_table_agate = run_query(test_compiled_code) %}
-        {% set test_row_dicts = elementary.agate_to_dicts(test_table_agate) %}
+{% macro get_test_sample(flatten_test_node) %}
+    {% set test_sample = graph["elementary"]["test_samples"].get(flatten_test_node.unique_id) %}
+    {% if not test_sample %}
+      {{ return([]) }}
```

Comment:
Doesn't it make more sense to store `null` if the test errored?
Just like how `test_results_query` and `failures` are `null`.